Import/SampleForLabelsInReports/Import/Method/Part Report With Label.xml

```diff
@@ -1,38 +1,46 @@
+<AML>
  <Item type="Method" id="394955B7C7DE4AEEAC8B0DFD903C16FD" action="add">
   <classification>/*</classification>
   <execution_allowed_to keyed_name="World" type="Identity">A73B655731924CD0B027E4F4D5FCC0A9</execution_allowed_to>
   <method_code><![CDATA[    var innovator = new Innovator();
-	var queryItem = new Item("Part","do_l10n");
-	queryItem.setAttribute("initial_action","get");
-	queryItem.setAttribute("select","item_number,name,make_buy,cost");
-	var partResults = queryItem.apply();
-	
-	for (var i=0; i<partResults.getItemCount(); i++){
-	 var thisPart = partResults.getItemByIndex(i);
-	 var partMakeBuy = "";
-	 partMakeBuy = thisPart.getProperty("make_buy");
-	 
-     var queryListItem = new Item("List","do_l10n");
-	 queryListItem.setAttribute("initial_action","get");
-     queryListItem.setAttribute("select","name");
-     queryListItem.setProperty("name", "Make Buy");
-     
-     var relatedItem = new Item("Value","do_l10n");
-     relatedItem.setAttribute("select","label,value");
-     relatedItem.setAttribute("initial_action","get");    
-	 relatedItem.setProperty("value", partMakeBuy);
-	 queryListItem.addRelationship(relatedItem);
-	 var listWithLabelValue = queryListItem.apply();
-	 var relatedResults = listWithLabelValue.getRelationships();
-	 var firstResult = relatedResults.getItemByIndex(0);
-	 var partMakeBuyLabel = firstResult.getProperty("label");
-	 thisPart.setProperty("partMakeBuyLabel", partMakeBuyLabel);
-	}
-	var report = innovator.getItemByKeyedName("Report","Part Report With Label");
-	var style = report.getProperty("xsl_stylesheet");
-	var html = partResults.applyStylesheet(style,"text");
-	return html;]]></method_code>
+var queryItem = new Item("Part", "get");
+queryItem.setAttribute("select", "item_number,name,make_buy,cost");
+var partResults = queryItem.apply();
+var dt = new Date(); 
+var d = dt.toLocaleString(navigator.language, {timeZoneName: 'short'});
+var ts = "Generated on: " + d;
+
+for (var i = 0; i < partResults.getItemCount(); i++) {
+    var thisPart = partResults.getItemByIndex(i);
+    var partMakeBuy = "";
+    partMakeBuy = thisPart.getProperty("make_buy");
+    //blank out error items and continue loop
+    if (partMakeBuy === null || partMakeBuy === "" || partMakeBuy === undefined){continue;}
+    var queryListItem = new Item("List", "get");
+    queryListItem.setAttribute("select", "name");
+    queryListItem.setProperty("name", "Make Buy");
+    var relatedItem = new Item("Value", "get");
+    relatedItem.setAttribute("select", "label,value");
+    relatedItem.setProperty("value", partMakeBuy);
+    relatedItem.apply();
+    queryListItem.addRelationship(relatedItem);
+    var listWithLabelValue = queryListItem.apply();
+    var relatedResults = listWithLabelValue.getRelationships();
+    var firstResult = relatedResults.getItemByIndex(0);
+    var partMakeBuyLabel = firstResult.getProperty("label");
+    thisPart.setProperty("partMakeBuyLabel", partMakeBuyLabel);
+    if (i === 0)
```

Comment:
Use `partResults.getItemByIndex(0);` outside the loop instead of executing this check on every iteration.
https://github.com/ArasLabs/list-label-xslt-sample/blob/c0f1f65cc58acc2573eb1a13426a59cd0392092d/Import/SampleForLabelsInReports/Import/Method/Part%20Report%20With%20Label.xml#L32-L35